Introduce the new Outro page for Chapter-1

shaavan opened the pull request:
Resolves #262

**Screenshot:**

<img width="1505" alt="image" src="https://user-images.githubusercontent.com/85434418/224082560-8adcd55b-efe7-4f32-8244-123bdebd6436.png">

shaavan commented:
Here's the preview link:

https://saving-satoshi-rjm58k58b-savingsatoshi.vercel.app/en/chapters/chapter-1/outro

benalleng commented:
tACK!
> Yes, the page background has a different gradient, that's the only change. You can ignore this for now.

I know the color change is shelved for now so other than that it looks good!

benalleng reviewed (commented):
Looks good! I wonder if making some minor changes to the `Introduction.tsx` template would allow us to not have to create a new template as they are almost identical.

GBKS commented on ui/chapter/Outro.tsx:
```diff
@@ -0,0 +1,49 @@
+        <div className="flex shrink basis-1/2">
+          <div className="flex flex-col gap-10 px-[15px] py-8 lg:px-10">
+            <div className="intro font-nunito text-xl text-white">
+              {children}
+            </div>
+            <div>
+              <Button href={next}>{btnText}</Button>
```
Yes, ideally it's the same template.

shaavan replied:
Resolved in 8ff2461a73207ac7c84677761fa6c236575d86eb

GBKS commented:
@ecurrencyhodler has a sharp eye. @shaavan the issue contains a link directly to the export-ready image in Figma that you need here. It should export to 1440x1440px.

shaavan commented:
>  the issue contains a link directly to the export-ready image in Figma that you need here. It should export to 1440x1440px.

I used the [export-ready image](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2323%3A25493&t=1WVTKCVc1kPVBDfc-1) to create the jpeg. However, I imported it with the `1x` format creating a 720px*720px image. That's why the image was not sharp!

Thanks, @ecurrencyhodler, for noticing the issue, and @GBKS, for pointing the correct dimensions for the export image.

GBKS commented:
@shaavan did you just press the export button or get the image out another way?

<img width="265" alt="image" src="https://user-images.githubusercontent.com/695901/224270020-ba0b2d90-1d7c-4c58-88fd-568f24f734d4.png">

shaavan commented:
Updated from 23ebaa04745e4bae76985aeb327c6fc25f8836ce to e0595bb723ad71f0a668a85302c5a93f241be4fe ([pr266.01](https://github.com/shaavan/saving-satoshi/commits/pr266.01) -> [pr266.02](https://github.com/shaavan/saving-satoshi/commits/pr266.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr266.01..pr266.02))
Addressed @ecurrencyhodler and @benalleng suggestions.

- Introduced a shared UI template between the **Outro** page and **Introduction.tsx.**
- Replaced the outro image with its crispier version.

shaavan commented:
@GBKS 

>  did you just press the export button or get the image out another way?

By pressing the export button. However, I selected dimensions to be `1x`. Now in the latest update, I have mended the image size/clarity discrepancy.

ecurrencyhodler commented:
tACK! Nice and clear. Let's get a code review in then merge.
Here's the preview link: https://saving-satoshi-1jevrj33p-saving-satoshi.vercel.app/en/chapters/chapter-1/outro

benalleng commented on content/lessons/outro/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import { useTranslations } from 'hooks'
+import { TextImage } from 'ui'
+
+export default function Outro({ lang }) {
```
Should we place Outro in the `/done` folder?

shaavan replied:
The `done` page symbolizes the end of the demo, whereas the `outro` page symbolizes the end of Chapter-1.
So they serve different purposes, and IMO is better to keep them in separate folders.

However, if you think it is better to keep them together, I would love to listen to your take on this.

benalleng replied:
Ok! That makes sense to me!

shaavan commented on ui/common/TextImageDisplay.tsx:
```diff
@@ -0,0 +1,69 @@
+        <div className="flex shrink basis-1/2">
+          <div className="flex flex-col gap-10 px-[15px] py-8 lg:px-10">
+            <div className="intro text-white">
+              {title && <h1 className="font-cbrush text-5xl">{t(title)}</h1>}
+              {subtitle && (
+                <h2 className="mb-2 pt-3 font-nunito text-xl font-black">
+                  {t(subtitle)}
+                </h2>
+              )}
```
Yes! I observed the modified files in #263. And I saw the Introduction.tsx doesn't display the title and subtitle.
Let me update the template accordingly!

shaavan replied:
Addressed in 8ff2461a73207ac7c84677761fa6c236575d86eb

shaavan commented:
Updated from e0595bb723ad71f0a668a85302c5a93f241be4fe to 8ff2461 ([pr266.02](https://github.com/shaavan/saving-satoshi/commits/pr266.02) -> [pr266.03](https://github.com/shaavan/saving-satoshi/commits/pr266.03))
Addressed @GBKS and @benalleng comments

Changes:

1. Rebased over the master.
2. Made appropriate changes to the `TextImageDisplay` file to accommodate for latest updates to master
3. Renamed `TextImageDisplay` -> `TextImageLayout`
4. Renamed _chapter-1-outro.jpg_ to _copter-interior-upscaled.jpg_ as per [export-ready image](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2323%3A25493&t=1WVTKCVc1kPVBDfc-1).

shaavan commented:
@GBKS

> just to clarify, by export-ready I mean both the file name and export settings are exactly as you should need them. No need to adjust.

The name of the export-ready image is `copter-interior-upscaled 1`, but I have modified it to `copter-interior-upscale`. Couldn't help myself from removing that extra _1_ 😄

@benalleng 

> With the Introduction.tsx and Outro.tsx now using the same template, can Introduction.tsx safely be removed?

I think Introduction.tsx is essential for now because it parses the `chapter` and `intro` metadata along with displaying the UI elements. So removing this file will increase complexity somewhere else.

shaavan commented on ui/chapter/Outro.tsx:
```diff
@@ -0,0 +1,49 @@
+import { Button } from 'shared'
+import { useLang, useTranslations } from 'hooks'
+
+export default function Outro({
```
Addressed in 8ff2461a73207ac7c84677761fa6c236575d86eb

benalleng reviewed (approved):
It LGTM I just want a final review. I am not at my desk to do a final review though and a want to make sure I am not a blocker.

GBKS commented:
@shaavan ah, I think I understand now where the misunderstanding comes from. The image is inside a frame. The frame is what you want to export. The frame is necessary sometimes to adjust cropping. 
![image](https://user-images.githubusercontent.com/695901/224488193-8743b636-bf3e-498e-b843-67c64178c5d1.png)

ecurrencyhodler commented:
@shaavan looks like there's a merge conflict: "content/lessons/holocat/index.tsx"

ecurrencyhodler commented:
Okay just as a heads up, once you resolve the remaining item, we can merge. Then we can merge #280 and #279. Then we'll only have one more PR to settle before tomorrow.

vinayaksh42 reviewed (changes requested):
@shaavan you will have to rebase your PR, in a recent PR #278 I added a [success page](https://savingsatoshi.com/en/chapters/chapter-1/genesis-4) for the completion of Challenge 1 which is using the `IntroductionLayout.tsx` which you renamed to `TextImageLayout.tsx`. You will have to change the component name for the success page and the location of `holocat` file has been shifted to `content/lesson/genesis/genesis-1`.

The navbar title currently displays `challenge 4` which is a bit wrong as we only have 2 challenges in chapter-1. if possible can you change it to display either `challenge 2` or display only chapter-1 (same as the [`done` page](https://savingsatoshi.com/en/chapters/chapter-1/done)). for making it display `challenge 2` you will have to add it to the `transacting` challenge grouping.
<img width="269" alt="Screenshot 2023-03-12 at 4 16 51 AM" src="https://user-images.githubusercontent.com/54861487/224515164-1116f1e7-1475-4d6f-be26-823dd0eaae2c.png">

shaavan commented:
Updated from 8ff2461 to c9a29362a7844b820b59f645686405956d0a28d3 ([pr266.03](https://github.com/shaavan/saving-satoshi/commits/pr266.03) -> [pr266.04](https://github.com/shaavan/saving-satoshi/commits/pr266.04))
Addressed @GBKS @ecurrencyhodler @vinayaksh42 comments

- Rebased over master and resolved merge conflicts.
- Imported the correct Figma frame for creating the image.
- Made the outro page the successor of **challenge 2** (`transacting-4` page) to display the correct challenge number in the navbar.
